packages/components/src/utils/orderedListSchemaBuilder.ts

```diff
@@ -0,0 +1,28 @@
+import { Type, type TSchema } from "@sinclair/typebox"
+
+export const orderedListSchemaBuilder = <T extends TSchema>(
+  listItemSchema: T,
+) =>
+  Type.Object(
+    {
+      type: Type.Literal("orderedList"),
+      attrs: Type.Optional(
+        Type.Object({
+          start: Type.Optional(
+            Type.Number({
+              title: "Starting number",
+              description: "The number to start the ordered list at",
+            }),
+          ),
+        }),
+      ),
+      content: Type.Array(listItemSchema, {
```

Comment:
wouldn't the list item schema here allow content to be `paragraph | unorderedList` in addition to `orderedList`?

Reply:
Hmm yup that's the intention. OrderedList takes in only a ListItem, which itself takes in Paragraph, OrderedList or UnorderedList.